src/staticdata.c

```diff
@@ -1249,6 +1250,9 @@ static void jl_write_values(jl_serializer_state *s) JL_GC_DISABLED
             ios_write(s->s, (char*)v, sizeof(void*) + jl_string_len(v));
             write_uint8(s->s, '\0'); // null-terminated strings for easier C-compatibility
         }
+        else if (jl_is_foreign_type(t) == 1) {
+            jl_error("Foreign Objects cannot be serialized");
+        }
```

Comment:
The corresponding code in `dump.c` has this:
```
        if (jl_is_foreign_type(t)) {
            jl_error("Cannot serialize instances of foreign datatypes");
        }
```
Perhaps these should be aligned?